src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java

```diff
@@ -12,11 +15,29 @@ public ActorDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Actor add(Actor actor) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(actor);
+            transaction.commit();
+        } catch (Exception e) {
+            transaction.rollback();
+            throw new DataProcessingException("can't add actor data: " + actor, e);
+        } finally {
+            session.close();
+        }
```

Comment:
A few points here.
 - `transaction` can be null.  Make rollback with null check.
```
            if (transaction != null) {
                transaction.rollback();
            }
```
 - Same for session
 ```
             if (session != null) {
                session.close();
            }
 ```
- optional: We can reduce code with using try-with-resources:
```
try (Session session = ...) {
  try {
    transaction = ....
    } catch (Exception e) {
     ...
    }
}

```